packages/core/src/logger/logger.ts

```diff
@@ -1,144 +1,196 @@
+    this.log(LogLevel.DEBUG, message, data);
+  }
+
+  public info(data: Record<string, any> | Error): void;
+  public info(message: string, data?: Record<string, any> | Error): void;
+  public info(message: any, data?: Record<string, any> | Error): void {
+    this.log(LogLevel.INFO, message, data);
+  }
+
+  public warning(data: Record<string, any> | Error): void;
+  public warning(message: string, data?: Record<string, any> | Error): void;
+  public warning(message: any, data?: Record<string, any> | Error): void {
+    this.log(LogLevel.WARNING, message, data);
+  }
+
+  public error(data: Record<string, any> | Error): void;
+  public error(message: string, data?: Record<string, any> | Error): void;
+  public error(message: any, data?: Record<string, any> | Error): void {
+    this.log(LogLevel.ERROR, message, data);
+  }
+
+  public fatal(data: Record<string, any> | Error): void;
+  public fatal(message: string, data?: Record<string, any> | Error): void;
+  public fatal(message: any, data?: Record<string, any> | Error): void {
+    this.log(LogLevel.ERROR, message, data);
+  }
+
+  public log(level: LogLevel | number, data: Record<string, any> | Error): void;
+  public log(level: LogLevel | number, message: string, data?: Record<string, any> | Error): void;
+  public log(
+    level: LogLevel | number,
+    messageOrDataOrError: Record<string, any> | Error | string,
+    maybeDataOrError?: Record<string, any> | Error,
   ): void {
-    const tag = this.context.getStore();
-
-    if (tag) {
-      this.queueManager.log(
-        tag,
-        pluginOrLevel,
-        levelOrMessage,
-        messageOrParams,
-        paramsOrData,
-        maybeData,
-      );
-    } else if (this.globalLogger) {
-      this.globalLogger.log(
-        pluginOrLevel,
-        levelOrMessage,
-        messageOrParams,
-        paramsOrData,
-        maybeData,
-      );
+    let error: Error | undefined;
+    let data: Record<string, any> | undefined;
+    let message: string | undefined;
+    if (typeof messageOrDataOrError === 'string') {
+      message = messageOrDataOrError;
+      if (maybeDataOrError instanceof Error) {
+        error = maybeDataOrError;
+      } else {
+        data = maybeDataOrError;
+      }
+    } else if (messageOrDataOrError instanceof Error) {
+      error = messageOrDataOrError;
+    } else {
+      data = messageOrDataOrError;
     }
+
+    this.add({
+      level,
+      message,
+      data,
+      error,
+    });
   }
 
-  setId(id: string): void {
-    const tag = this.context.getStore();
-    tag && this.queueManager.setQueueId(tag, id);
+  public add(entry: Omit<LogEntry<Partial<TContext>, TGlobalContext>, 'context' | 'ts'>): void {
+    const context: Partial<TContext> = this.asyncStorage.getStore() || {};
+
+    this.logHandlers.forEach((logHandler) => {
+      logHandler.log({
+        level: entry.level,
+        context: JSON.parse(JSON.stringify({ ...context, ...this.globalContext })),
```

Comment:
použij `v8.deserialize(v8.serialize({ ... }))`, je to rychlejší a umí to věci, který JSON neumí (cyklický reference, Date, ...)

Reply:
https://nodejs.org/api/v8.html#serialization-api

Reply:
done